Gems/Atom/Feature/Common/Code/Tests/SparseVectorTests.cpp

```diff
@@ -31,6 +31,18 @@ namespace UnitTest
         bool c = DefaultValueC;
     };
 
+    struct TestDataWithDestructor
+    {
+        bool m_destroyed = false;
+        size_t m_value = 100;
+
+        ~TestDataWithDestructor()
+        {
+            EXPECT_FALSE(m_destroyed);
+            m_destroyed = true;
```

Comment:
What is the point of setting this to `true`?
After the destructor finishes, it is not safe to read its value again so setting the m_destroyed variable doesn't do anything.

Reply:
Ah, I see this logic is to detect a second call to the destructor.

This is fine then.

Reply:
yeah, in this particular context we know the memory is stored inside the SparseVector. It is possible this memory will get overwritten by the linked list of empty slots, but at most one of those entries would have a value of 0 written to it, so other entries would still interpret `m_destroyed` as true.